osm2streets/src/transform/collapse_intersections.rs

```diff
@@ -51,7 +48,7 @@ fn should_collapse(r1: OriginalRoad, r2: OriginalRoad, streets: &StreetNetwork)
     // a bizarre example. These are actually blackholed, some problem with service roads.
     if road1.oneway_for_driving().is_some()
         && road2.oneway_for_driving().is_some()
-        && r1.i2 == r2.i2
+        && road1.id.i2 == road2.id.i2
```

Comment:
Maybe introducing `start_intersection()` and `end_intersection()` (or whetever the naming should be) would be a worthwhile stepping stone towards opaque ids.